### ISSUE TITLE
Surface Book Working Clover Folder 10.15.4

### ISSUE THREAD

**Issue by carloscarval** [bigsadan/surface-pro-4-hackintosh#67]
Hi, I'm passing by to drop another Clover Folder for Surface book on 10.15.4 MacOS.
I'm doing this because I downloaded from another issue here and I couldn't boot with the Config.plist. 
I had to apply two patches for graphics and finally everything worked fine.
Graphic works, battery used is the one in the keyboard, touchpad works, seems everything fine.
Did the clean installation from USB drive.
I hope it works for you too.
[CLOVER.zip](https://github.com/bigsadan/surface-pro-4-hackintosh/files/4617743/CLOVER.zip)

**Comment by gityasss** [comment]
hi there
does touchpad works with gestures and shows in system preferences ?

**Comment by carloscarval** [comment]
> hi there
> does touchpad works with gestures and shows in system preferences ?

It does for me.

**Comment by gityasss** [comment]
have you applied any sudo command in terminal or just copied the two kext to other folder ?

**Comment by carloscarval** [comment]
> have you applied any sudo command in terminal or just copied the two kext to other folder ?

No.. not any sudo command.
Just installed with some other CLOVER folder. Then, just changed the CLOVER to the way it is now.

**Comment by gityasss** [comment]
yes that clover folder you used is mine , i will try with your clover folder and do a clean install 
because when i put the two kext in other folder it gives me kernel panic when i reboot

**Comment by gityasss** [comment]
do you have surface book with performance base (nvidia graphic 940m) ?

**Comment by carloscarval** [comment]
> do you have surface book with performance base (nvidia graphic 940m) ?

Yes, I do, but only intel graphics works.. 
I’ve been trying to “merge” the two batteries in order to use them both in MacOS. Since everything that I’ve seen, people only use the keyboard or the screen battery. And I realized That the CLOVER folder that I posted is not really that good.
The battery seems to be working fine(the one in the keyboard), but sometimes it doesn’t.
And during boot, using verbose, I could see some problems about battery Information.
I’m saying that because I said the battery was working properly and it’s not.

**Comment by gityasss** [comment]
![20200513_201212](https://user-images.githubusercontent.com/63503032/81872855-27801980-956a-11ea-8ee0-83610d6bcf55.jpg)
Hi carlos
i did a fresh 10.15.4 install with your clover folder but when i reach to the second reboot of install it gives me this kernel panic no matter if i choose configis or config i cant get past that error
if i delete the two voodooi2c kext then i cant get to desktop successfully , but trackpad has no gestures and doesnt show on system preferences !

**Comment by carloscarval** [comment]
Well, I don’t know much about these things. So I did a lot of brute force trying. I’d suggest you to try to Boot in safe mood and verbose (-v -x). And during that running Kext Utility. It rebuilds the kext cache automatically. Just open, type your password and wait. Maybe that will help in order to make the touchpad work.

**Comment by carloscarval** [comment]
And actually, the same happened to me. Kernel panic when trying to boot with Voodoo kext injected. I can't only boot in safe mode :/

**Comment by gityasss** [comment]
> And actually, the same happened to me. Kernel panic when trying to boot with Voodoo kext injected. I can't only boot in safe mode :/

hi there
so how did you manage to have trackpad gestures as you said if you get kernel panic with the voodooi2c kext ?

**Comment by carloscarval** [comment]
> > And actually, the same happened to me. Kernel panic when trying to boot with Voodoo kext injected. I can't only boot in safe mode :/
> 
> hi there
> so how did you manage to have trackpad gestures as you said if you get kernel panic with the voodooi2c kext ?

I don't know.. It was working properly, and after some changes in DSDT, it stopped working.. I've been struggling to get both batteries working as one, maybe some of that messed up the touchpad gestures too.

**Comment by gityasss** [comment]
> > > And actually, the same happened to me. Kernel panic when trying to boot with Voodoo kext injected. I can't only boot in safe mode :/
> > 
> > 
> > hi there
> > so how did you manage to have trackpad gestures as you said if you get kernel panic with the voodooi2c kext ?
> 
> I don't know.. It was working properly, and after some changes in DSDT, it stopped working.. I've been struggling to get both batteries working as one, maybe some of that messed up the touchpad gestures too.

yes.. maybe it has a relation with batteries ,anyway if you get it working back again , let me know 
greetings

**Comment by carloscarval** [comment]
> > > > And actually, the same happened to me. Kernel panic when trying to boot with Voodoo kext injected. I can't only boot in safe mode :/
> > > 
> > > 
> > > hi there
> > > so how did you manage to have trackpad gestures as you said if you get kernel panic with the voodooi2c kext ?
> > 
> > 
> > I don't know.. It was working properly, and after some changes in DSDT, it stopped working.. I've been struggling to get both batteries working as one, maybe some of that messed up the touchpad gestures too.
> 
> yes.. maybe it has a relation with batteries ,anyway if you get it working back again , let me know
> greetings

Hey, did you solve your problems with touchpad gestures? I don't know what happened. But I can't boot with Voodoo injected in any way.
I tried clean install of 10.15.4, 10.15.3, 10.14.6. And in none of them I could boot and make touchpad be recognized or make the system boot with Voodoo injected.
And it's super weird, because I feel like I tested every DSDT in repository and other, with different Clover folder and I just don't know what to do any more?
Did you have more luck than me?

**Comment by basser2** [comment]
> Hi, I'm passing by to drop another Clover Folder for Surface book on 10.15.4 MacOS.
> I'm doing this because I downloaded from another issue here and I couldn't boot with the Config.plist.
> I had to apply two patches for graphics and finally everything worked fine.
> Graphic works, battery used is the one in the keyboard, touchpad works, seems everything fine.
> Did the clean installation from USB drive.
> I hope it works for you too.
> [CLOVER.zip](https://github.com/bigsadan/surface-pro-4-hackintosh/files/4617743/CLOVER.zip)

my surface book is i7,8g,intel520,gtx965m,but I cannot use your clover to boot,could you help me?

**Comment by beevu96dn** [comment]
> cuốn sách bề mặt của tôi là i7,8g, intel520, gtx965m, nhưng tôi không thể sử dụng cỏ ba lá của bạn để khởi động, bạn có thể giúp tôi không?



I have a way to make it work, please delete DSDT.aml so it will help you to use normally in addition to not receiving battery. The trackpad is still active IGPU is still active. I also don't know if it's due to DSDT or not but with DSDT I can't boot
